service/controller/v13/resource/s3object/desired.go

```diff
@@ -29,6 +29,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	clusterID := key.ClusterID(customObject)
 
+	_, err = r.encrypter.EncryptionKey(ctx, customObject)
 	if IsKeyNotFound(err) {
 		// we can get here during deletion, if the key is already deleted we can safely exit.
 		return output, nil
```

Comment:
The zero value of a map is `nil`. I always find these things dangerous because you never know where `output` might be filled above by somebody who is not away of the full implementation.

Reply:
Totally agree, we should return nil instead.